src/System.IO.Pipelines/src/System/IO/Pipelines/PipeReader.cs

```diff
@@ -89,7 +89,38 @@ public static PipeReader Create(Stream stream, StreamPipeReaderOptions readerOpt
         /// <summary>
         /// Asynchronously reads the bytes from the <see cref="PipeReader"/> and writes them to the specified stream, using a specified buffer size and cancellation token.
         /// </summary>
-        /// <param name="destination">The stream to which the contents of the current stream will be copied.</param>
+        /// <param name="destination">The <see cref="PipeWriter"/> to which the contents of the current stream will be copied.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
+        /// <returns>A task that represents the asynchronous copy operation.</returns>
+        public virtual Task CopyToAsync(PipeWriter destination, CancellationToken cancellationToken = default)
+        {
+            if (destination == null)
+            {
+                throw new ArgumentNullException(nameof(destination));
+            }
+
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return Task.FromCanceled(cancellationToken);
+            }
+
+            static async ValueTask WriteAsync(PipeWriter destination, ReadOnlyMemory<byte> memory, CancellationToken cancellationToken)
+            {
+                FlushResult result = await destination.WriteAsync(memory, cancellationToken).ConfigureAwait(false);
+
+                if (result.IsCanceled)
+                {
+                    ThrowHelper.ThrowOperationCanceledException_FlushCanceled();
+                }
+            }
+
+            return CopyToAsyncCore(destination, WriteAsync, cancellationToken);
```

Comment:
This is going to allocate a delegate for WriteAsync on each call; presumably that's not desirable.  In which case you'd be better off not using a local function and instead just using a lambda:
```C#
return CopyToAsyncCore(destination, (destination, memory, cancellationToken) =>
{
    FlushResult result = await destination.WriteAsync(memory, cancellationToken).ConfigureAwait(false);
    if (result.IsCanceled)
    {
        ThrowHelper.ThrowOperationCanceledException_FlushCanceled();
    }
}, cancellationToken);
```

Reply:
We never fixed that!?

Reply:
https://github.com/dotnet/roslyn/issues/5835